### ISSUE TITLE
`uniqueConstraints` are ignored in `#[Table]`

### ISSUE THREAD

**Issue by kiler129** [doctrine/doctrine2#10422]
### Bug Report
|    Q        |   A
|------------ | ------
| BC Break    | no
| Version     | 2.14.0

#### Summary
When configuring an entity with unique constraints, the only way it work is when `UniqueConstraint` attribute is attached to the class. It doesn't seem to work when a parameter to `Table` is used. See reproducer.

#### Current behavior
No indexes are created when `Table(uniqueConstraints: ...)` is used.

#### How to reproduce
An entity defined like that will not create any indexes, according to `doctrine:schema:update`, while passing `doctrine:schema:validate` check:
```php
#[ORM\Table(
    name: 'cohort_animals',
    uniqueConstraints: [
        new ORM\UniqueConstraint(name: 'cohort_animal_uniq', columns: ['cohort_id', 'participant_id'])
    ]
)]
class CohortAnimal {...}
```

Changing the entity to use `UniqueConstraint` on the class directly solves the issue:
```php
#[ORM\UniqueConstraint(name: 'cohort_animal_uniq', columns: ['cohort_id', 'participant_id'])]
class CohortAnimal {...}
```

#### Expected behavior
Ideally, the first syntax should be supported. If it isn't desired an schema validation error should occur. This may be related to #8676

**Comment by derrabus** [comment]
> No indexes are created when `Table(uniqueConstraints: ...)` is used.

Correct. This parameter is ignored when using attributes, it's a leftover from the annotations mapping.

> Changing the entity to use `UniqueConstraint` on the class directly solves the issue:

Correct, this is how `UniqueConstraint` is documented to work.

> Ideally, the first syntax should be supported.

No, that's far from what I'd call ideal. We don't need two ways to achieving the same thing.

> If it isn't desired an schema validation error should occur.

PR welcome. 🙂 

> This may be related to #8676

It isn't.
